### PR TITLE
[Documentation] Write more How To

### PR DESCRIPTION
**Pull request by chibenwa**
Screenshot from 2021-07-30 17-06-34

The following PR aims at providing more hands on example on how to write custom components and load them into Apache James, including:

 - SMTP hooks
 - SMTP commands
 - WebAdmin routes

We also demonstrate how to assemble a tailor made James server.

For each of them we include an example and a how-to on the WebSite.

Also, a README pointing to the how to had been added for existing examples, providing more context. A README lists the various examples, easing discovery of examples on GitHub.

Finally we fix inaccurate JavaDoc for some SMTP hooks.

### CODE REVIEW

File: src/homepage/howTo/custom-smtp-hooks.html [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+    <!-- Introduction -->
+    <section id="intro" class="main special">
+        <div class="">
+            <div class="content align-left">
+                <header class="major">
+                    <h1><b>Configure Custom SMTP hooks</b></h1>
+                </header>
+
+                <p>
+                    The current project demonstrate how to write custom behaviours for Apache James SMTP server
```
Comment (Arsnael) [inline src/homepage/howTo/custom-smtp-hooks.html:35]:
```suggestion
                    The current project demonstrates how to write custom behaviors for Apache James SMTP server
```

File: src/homepage/howTo/custom-smtp-hooks.html [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+                    by the means of SMTP hooks.
+                </p>
+
+                <p>
+                    SMTP hooks allows integrating third party systems with the SMTP stack, allows writing additional SMTP extensions, for
```
Comment (Arsnael) [inline src/homepage/howTo/custom-smtp-hooks.html:40]:
```suggestion
                    SMTP hooks allow integrating third party systems with the SMTP stack and writing additional SMTP extensions, for
```

File: examples/custom-smtp-command/README.md [hunk 3]
```diff
@@ -0,0 +1,151 @@
+# Creating your own SMTP commands
+
+Read this page [on the website](http://james.apache.org/howTo/custom-smtp-commands.html).
+
+The current project demonstrate how to write custom commands for Apache James SMTP server.
```
Comment (Arsnael) [inline examples/custom-smtp-command/README.md:5]:
```suggestion
The current project demonstrates how to write custom commands for Apache James SMTP server.
```

File: src/homepage/howTo/custom-smtp-commands.html [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+    <!-- Introduction -->
+    <section id="intro" class="main special">
+        <div class="">
+            <div class="content align-left">
+                <header class="major">
+                    <h1><b>Configure Custom SMTP commands</b></h1>
+                </header>
+
+                <p>
+                    The current project demonstrate how to write custom commands for Apache James SMTP server.
```
Comment (Arsnael) [inline src/homepage/howTo/custom-smtp-commands.html:35]:
```suggestion
                    The current project demonstrates how to write custom commands for Apache James SMTP server.
```

File: src/homepage/howTo/custom-webadmin-routes.html [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+    <!-- Introduction -->
+    <section id="intro" class="main special">
+        <div class="">
+            <div class="content align-left">
+                <header class="major">
+                    <h1><b>Configure Custom WebAdmin routes</b></h1>
+                </header>
+
+                <p>
+                    The current project demonstrate how to write custom webadmin routes for Apache James. This enables writing new
```
Comment (Arsnael) [inline src/homepage/howTo/custom-webadmin-routes.html:35]:
```suggestion
                    The current project demonstrates how to write custom webadmin routes for Apache James. This enables writing new
```

File: examples/custom-james-assembly/README.md [hunk 6]
```diff
@@ -0,0 +1,154 @@
+# Assemble a James server tailored to your needs
+
+Read this page [on the website](http://james.apache.org/howTo/custom-james-assembly.html).
+
+The current project demonstrate how to write a custom assembly in order to write your
```
Comment (Arsnael) [inline examples/custom-james-assembly/README.md:5]:
```suggestion
The current project demonstrates how to write a custom assembly in order to write your
```

File: src/homepage/howTo/custom-james-assembly.html [hunk 7]
```diff
@@ -0,0 +1,190 @@
...
+            <div class="content align-left">
+                <header class="major">
+                    <h1><b>Write Custom James server assembly</b></h1>
+                </header>
+                <p>
+                    Find this example on <a href="https://github.com/apache/james-project/tree/master/examples/custom-james-assembly">GitHub</a>.
+                </p>
+
+                <p>
+                    The current project demonstrate how to write a custom assembly in order to write your
```
Comment (Arsnael) [inline src/homepage/howTo/custom-james-assembly.html:38]:
```suggestion
                    The current project demonstrates how to write a custom assembly in order to write your
```

File: examples/README.md [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+ to all the Listeners that had been registered before. After receiving events, listeners retrieve information from the 
+ events then execute their business (Indexing emails in ElasticSearch, updating quota of users, detecting spam emails...)
+ 
+**Mailbox Listeners** allow customizing the behaviour of James used as a Mail Delivery Agent (MDA). 
+
+[This example](custom-listeners) shows how to write such components!
+
+## Configure Custom SMTP hooks
+
+SMTP hooks allows integrating third party systems with the SMTP stack, allows writing additional SMTP extensions, for
```
Comment (Arsnael) [inline examples/README.md:33]:
```suggestion
SMTP hooks allow integrating third party systems with the SMTP stack, allows writing additional SMTP extensions, for 
```

File: examples/README.md [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+
+## Configure Custom SMTP commands
+
+This subproject demonstrates how to write custom commands for Apache James SMTP server. 
+
+[This example](custom-smtp-command) shows how to write additional SMTP commands!
+
+## Configure Custom WebAdmin routes
+
+The current project demonstrate how to write custom webadmin routes for Apache James. This enables writing new
```
Comment (Arsnael) [inline examples/README.md:46]:
```suggestion
The current project demonstrates how to write custom webadmin routes for Apache James. This enables writing new 
```

File: examples/README.md [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+
+The current project demonstrate how to write custom webadmin routes for Apache James. This enables writing new 
+administrative features exposed over a REST API. This can allow you to write some additional features, make James 
+interact with third party systems, do advance reporting... 
+
+[This example](custom-webadmin-route) shows how to write additional Webadmin routes!
+
+## Write Custom James server assembly
+
+[This example](custom-james-assembly) demonstrate how to write a custom assembly in order to write your own tailor-made server.
```
Comment (Arsnael) [inline examples/README.md:54]:
```suggestion
[This example](custom-james-assembly) demonstrates how to write a custom assembly in order to write your own tailor-made server.
```

**Review by Arsnael — APPROVED** [review]

**Comment by chibenwa** [comment]
Just squashed

**Comment by chibenwa** [comment]
Was the website updated?

**Comment by Arsnael** [comment]
> Was the website updated?

How do you update a website based on code not merged? If that's really the process I find that rather strange...

**Comment by chibenwa** [comment]
https://james.staged.apache.org/james-project/3.7.0/community/website.html

**Comment by chibenwa** [comment]
Done by the way

**Pull request merged**